### PR TITLE
Add annotation that lets you skip validation

### PR DESCRIPTION
**Pull request by dmontagu**
Closes https://github.com/pydantic/pydantic/issues/1465

Selected Reviewer: @adriangb

### CODE REVIEW

**Review by Kludex — APPROVED** [review]

File: pydantic/annotated_arguments.py [hunk 1]
```diff
@@ -105,3 +105,16 @@ def __get_pydantic_core_schema__(
...
+    """
+    If this is applied as an annotation (e.g., via `x: Annotated[int, SkipValidation]`), validation will be skipped.
+
+    This can be useful if you want to use a type annotation for documentation/IDE/type-checking purposes,
+    and know that it is safe to skip validation for one or more of the fields.
+    """
+
+    @classmethod
+    def __get_pydantic_core_schema__(self, _source: Any, _handler: Any) -> core_schema.CoreSchema:
+        return core_schema.any_schema()
```
Comment (adriangb) [inline pydantic/annotated_arguments.py:120]:
```suggestion
    @classmethod
    def __get_pydantic_core_schema__(cls, _source: Any, _handler: Any) -> core_schema.CoreSchema:
        return core_schema.any_schema()
```

**Review by adriangb — APPROVED** [review]

**Comment by adriangb** [comment]
How does this work with JSON schema and serialization? Is there a world where we want:
- To preserve the JSON schema or serializer of the original type but skip validation?
- Let users specify a JSON schema or serializer to accompany this AnyValidator?

**Comment by samuelcolvin** [comment]
JSON schema should be be preserved, e.g. generated from the inner type. 

I think we should use the serialiser from the inner type too. We could add a switch for this, but it could come later.

**Comment by dmontagu** [comment]
@adriangb do you see how to make the JSON schema thing work in this way now that the "js_cs_override_schema" metadata thing is gone? I'm not sure what exactly the current signature is for json schema generation, or if there is a good way to handle this now.

@samuelcolvin is there a way to take a "regular" core schema and use it for serialization? Like, ideally I'd be able to do something like:
```python
    @classmethod
    def __get_pydantic_core_schema__(self, _source: Any, _handler: Any) -> core_schema.CoreSchema:
        schema = _handler(_source)
        return core_schema.any_schema(serialization=schema)
```
but I think serialization needs to specifically be a `SerSchema`. Maybe we need to do a rework to this.

Either way, I think this can wait until after v2, though @adriangb I think it would be good for us to make sure it's possible to handle the JSON schema for this without needing to make breaking changes. (I guess we can always go back to using new core schema metadata keys if we need to, so it doesn't need to be a breaking change, but.. I guess it would be better to avid that)

**Comment by dmontagu** [comment]
please review

**Comment by adriangb** [comment]
I think you’ll want `__prepare_pydantic_annotations__` so that you can still create a json schema.

**Comment by dmontagu** [comment]
please update

**Comment by dmontagu** [comment]
please review

File: pydantic/_internal/_generate_schema.py [hunk 2]
```diff
@@ -742,7 +742,7 @@ def _generate_parameter_schema(
         else:
             field = FieldInfo.from_annotated_attribute(annotation, default)
         assert field.annotation is not None, 'field.annotation should not be None when generating a schema'
-        source_type, annotations = annotation, field.metadata
+        source_type, annotations = field.annotation, field.metadata
```
Comment (dmontagu) [inline pydantic/_internal/_generate_schema.py:745]: even if there are issues merging the rest of this, we should probably make this fix either way.

**Review by adriangb — APPROVED** [review]

**Pull request merged**